src/async_reader/zarr_read_async.rs

```diff
@@ -73,13 +76,27 @@ impl<'a> ZarrReadAsync<'a> for ZarrPath {
         position: &'a [usize],
         cols: &'a [String],
         real_dims: Vec<usize>,
+        separators: HashMap<String, ChunkSeparator>,
     ) -> ZarrResult<ZarrInMemoryChunk> {
         let mut chunk = ZarrInMemoryChunk::new(real_dims);
         for var in cols {
             let s: Vec<String> = position.iter().map(|i| i.to_string()).collect();
-            let s = s.join(".");
-
-            let p = self.location.child(var.to_string()).child(s);
+            let separator = separators
```

Comment:
This is an actual bug fix, that needs to go in, unrelated to the async stream work.